Shellcheck failure since recent update causes nixos-anywhere test failure

My flake is moving from `856a2902156ba304efebd4c1096dbf7465569454` to `5e40e02978e3bd63c2a6a9fa6fa8ba0e310e747f` and now fails with the following when running nixos-anywhere and using disko.

```
error: builder for '/nix/store/kkv73jq43r5i1kwiwjizwbf34cqkwcjl-disko-destroy-format-mount.drv' failed with exit code 1;
       last 12 log lines:
       >
       > In /nix/store/fhk167wv3j36rnxihbahfsx45hk4ajzd-disko-destroy-format-mount line 7:
       >   for dev in '/dev/vdb'; do
       >              ^--------^ SC2041 (warning): This is a literal string. To run as a command, use $(..) instead of '..' . 
       >
       >
       > In /nix/store/fhk167wv3j36rnxihbahfsx45hk4ajzd-disko-destroy-format-mount line 25:
       > for dev in '/dev/vdb'; do
       >            ^--------^ SC2041 (warning): This is a literal string. To run as a command, use $(..) instead of '..' . 
       >
       > For more information:
       >   https://www.shellcheck.net/wiki/SC2041 -- This is a literal string. To run ...
       For full logs, run 'nix log /nix/store/kkv73jq43r5i1kwiwjizwbf34cqkwcjl-disko-destroy-format-mount.drv'.
error: 1 dependencies of derivation '/nix/store/6dgr3rj60zmhawmmq0p612j6xi0vhimh-nixos-test-driver-disko-lab-disko.drv' failed to build
```

I believe that this is referring to this loop
https://github.com/nix-community/disko/blob/5e40e02978e3bd63c2a6a9fa6fa8ba0e310e747f/lib/default.nix#L559-L562

Which appears to have been changed recently in this commit https://github.com/nix-community/disko/commit/94bc0f5bb0dff8b98597f1eb66a7156469932221 via this pull request https://github.com/nix-community/disko/pull/842

Hmm, you did correctly identify the location where this error occurs, but we're using [`escapeShellArgs`](https://noogle.dev/f/lib/escapeShellArgs) here, which usually doesn't add single-quotes around items unless needed.

```nix
  let
    selectedDisks = lib.escapeShellArgs (lib.catAttrs "device" (lib.attrValues devices.disk));
  in
  ''
    if [ "$1" != "--yes-wipe-all-disks" ]; then
      echo "WARNING: This will destroy all data on the disks defined in disko.devices, which are:"
      echo
      # shellcheck disable=SC2043
      for dev in ${selectedDisks}; do
        echo "  - $dev"
      done
      [...]
  ''
```

Are you using `follows` to override the nixpkgs input for disko? If so, which revision are you using?

--- a/lib/default.nix
+++ b/lib/default.nix
@@ -556,7 +556,7 @@ let
                 if [ "$1" != "--yes-wipe-all-disks" ]; then
                   echo "WARNING: This will destroy all data on the disks defined in disko.devices, which are:"
                   echo
-                  # shellcheck disable=SC2043
+                  # shellcheck disable=SC2043,2041
                   for dev in ${selectedDisks}; do
                     echo "  - $dev"
                   done

